src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -229,6 +205,41 @@ def get_custom_templatetags_libraries(self) -> list[str]:
             f"{prefix}.objects_api.json_tags",
         ]
 
+    def update_payment_status(self, submission: "Submission", options: dict) -> None:
+        config = ObjectsAPIConfig.get_solo()
+        assert isinstance(config, ObjectsAPIConfig)
+        config.apply_defaults_to(options)
+
+        if not options["payment_status_update_json"]:
+            logger.warning(
+                "Skipping payment status update because no template was configured."
+            )
+            return
+
+        context = {
+            "variables": get_variables_for_context(submission),
+            "payment": get_payment_context_data(submission),
+        }
+
+        updated_record_data = render_to_json(
+            options["payment_status_update_json"], context
+        )
+        updated_object_data = {
+            "record": {
+                "data": {"payment": updated_record_data},
```

Comment:
isn't it more logical to leave the form builders to specify the `payment` key rather than hardcoding it? IMO this should be part of the template.